ModelVale/ModelVale/ViewControllers/Main/DataViewController.m

```diff
@@ -6,26 +6,96 @@
 //
 
 #import "DataViewController.h"
+#import "UserDataCell.h"
+#import "UserDataSectionHeader.h"
+#import "ModelData.h"
+#import "ModelLabel.h"
+#import "TestTrainEnum.h"
+#import "UIViewController+PresentError.h"
 
-@interface DataViewController ()
-
+@interface DataViewController () <UICollectionViewDelegate, UICollectionViewDataSource>
+@property (weak, nonatomic) IBOutlet UICollectionView *userDataCollectionView;
 @end
 
 @implementation DataViewController
 
+// Need two section headers (or rather, num options in testTrainPickerView in AddData) for each label
 - (void)viewDidLoad {
     [super viewDidLoad];
-    // Do any additional setup after loading the view.
+    self.modelLabels = [NSMutableArray new];
+    self.userDataCollectionView.delegate = self;
+    self.userDataCollectionView.dataSource = self;
+    
+    ModelLabel* fakeLabel = [[ModelLabel new] initEmptyLabel:@"mountain" testTrainType:Train sectionInd:0];
+    UIImage* testImage = [UIImage imageNamed:@"mountain"];
+    ModelData* fakeData = [[ModelData new] initWithImage:testImage label:fakeLabel]; // Note: adds itself to the label passed
+    testImage = [UIImage imageNamed:@"rivermountain"];
+    fakeData = [[ModelData new] initWithImage:testImage label:fakeLabel];
+    [self.modelLabels addObject:fakeLabel];
+    // Add one mountain to the label "hill" and add to dvc
+    fakeLabel = [[ModelLabel new] initEmptyLabel:@"hill" testTrainType:Train sectionInd:1];
+    testImage = [UIImage imageNamed:@"snowymountains"];
+    fakeData = [[ModelData new] initWithImage:testImage label:fakeLabel];
+    [self.modelLabels addObject:fakeLabel];
+    [self.userDataCollectionView reloadData];
+}
+
+// Return the index of the section we're in given the row
```

Comment:
instead of commenting the function try making the name more clear so maybe getSectionsForRow:(NSInteger)row

Reply:
so by this the programmer now knows that he can uses that function to get the sections

Reply:
I was trying to figure out how to return two values, and since Obj-C doesn't have tuples I used an array